lib/broadway/options.ex

```diff
@@ -300,4 +318,20 @@ defmodule Broadway.Options do
     {:error,
      "expected :name to be an atom or a {:via, module, term} tuple, got: #{inspect(name)}"}
   end
+
+  def validate_batch_size(size) when is_integer(size) and size > 0, do: {:ok, size}
+
+  def validate_batch_size({func, _acc} = batch_splitter) when is_function(func) do
+    if :erlang.fun_info(func)[:arity] == 2 do
+      {:ok, batch_splitter}
+    else
+      {:error,
+       "expected batch sizing function has an arity of 2, got: #{inspect(:erlang.fun_info(func)[:arity])}\n"}
```

Comment:
```suggestion
       "expected `:batch_size` to include a function of 2, got: #{inspect(fun)}"
```

Reply:
I do not know why I must add the trailing \n to pass the test locally.  I do not see any difference on expected and actual messages shown in failed CI.

Reply:
Fixed as it's the captured function again.